src/morphological_tagger/MorphDictionary.cpp

```diff
@@ -0,0 +1,377 @@
+//  while (instance != NULL) {
+//    int instance_length = instance->size();
+//    for (int i = 0; i < instance_length; ++i) {
+//      int id;
+//      string form = instance->GetForm(i);
+//      if (!form_case_sensitive) {
+//        transform(form.begin(), form.end(), form.begin(), ::tolower);
+//      }
+//      int word_id = token_dictionary_->GetFormId(form);
+//      //CHECK_GE(word_id, 0);
+//
+//      id = tag_alphabet_.Lookup(instance->GetTag(i));
+//      CHECK_GE(id, 0);
+//
+//      // Insert new tag in the set of word tags, if it is not there
+//      // already. NOTE: this is inefficient, maybe we should be using a
+//      // different data structure.
+//      if (word_id >= 0) {
+//        vector<int> &tags = word_tags_[word_id];
+//        int j;
+//        for (j = 0; j < tags.size(); ++j) {
+//          if (tags[j] == id) break;
+//        }
+//        if (j == tags.size()) tags.push_back(id);
+//      }
+//    }
+//    delete instance;
+//    instance = static_cast<SequenceInstance*>(reader->GetNext());
+//  }
+//  reader->Close();
+//
+//  for (int i = 0; i < tag_alphabet_.size(); ++i) {
+//    unknown_word_tags_.push_back(i);
+//  }
+//  LOG(INFO) << "Number of unknown word tags: " << unknown_word_tags_.size();
+//}
+
+
+void MorphDictionary::CreateTagDictionary(MorphReader *reader) {
+  SequenceDictionary::CreateTagDictionary(static_cast<SequenceReader*> (reader));
+
+  LOG(INFO) << "Creating cpostag-morphtag dictionary...";
+  bool form_case_sensitive = FLAGS_form_case_sensitive;
+
+  // Go through the corpus and build the existing tags for each word.
+  cpostag_morphtags_.clear();
+  cpostag_morphtags_.resize(token_dictionary_->GetNumCPosTags());
+
+  reader->Open(pipe_->GetOptions()->GetTrainingFilePath());
+  MorphInstance *instance = static_cast<MorphInstance*>(reader->GetNext());
+  while (instance != NULL) {
+    int instance_length = instance->size();
+    for (int i = 0; i < instance_length; ++i) {
+      int id;
+      string cpostag = instance->GetCoarsePosTag(i);
+      int cpostag_id = token_dictionary_->GetCoarsePosTagId(cpostag);
+      //CHECK_GE(cpostag_id, 0);
+
+      id = tag_alphabet_.Lookup(instance->GetTag(i));
+      CHECK_GE(id, 0);
+
+      // Insert new tag in the set of word tags, if it is not there
+      // already. NOTE: this is inefficient, maybe we should be using a
+      // different data structure.
+      if (cpostag_id >= 0) {
+        vector<int> &tags = cpostag_morphtags_[cpostag_id];
+        int j;
+        for (j = 0; j < tags.size(); ++j) {
+          if (tags[j] == id) break;
+        }
+        if (j == tags.size()) tags.push_back(id);
+      }
+    }
+    delete instance;
+    instance = static_cast<MorphInstance*>(reader->GetNext());
+  }
+  reader->Close();
+
+  for (int i = 0; i < tag_alphabet_.size(); ++i) {
+    unknown_cpostag_morphtags_.push_back(i);
+  }
+  LOG(INFO) << "Number of unknown cpostag morphtags: " << unknown_cpostag_morphtags_.size();
+}
+
```

Comment:
Get rid of extra blank lines. Use only one blank line between function declarations.